radixdlt/src/main/java/com/radixdlt/consensus/VertexStore.java

```diff
@@ -194,18 +352,23 @@ private boolean addQC(QuorumCertificate qc) {
 			highestQC.set(qc);
 		}
 
-		QuorumCertificate highestCommitted = highestCommittedQC.get();
 		qc.getCommitted().ifPresent(vertexMetadata -> {
-			if (!highestCommitted.getCommitted().isPresent()
-				|| highestCommitted.getCommitted().get().getStateVersion() < vertexMetadata.getStateVersion()) {
+			QuorumCertificate highestCommitted = highestCommittedQC.get();
+			Optional<VertexMetadata> highest = highestCommitted.getCommitted();
+			if (highest.isPresent()) {
+				if (highest.get().getView().compareTo(vertexMetadata.getView()) < 0) {
+					this.highestCommittedQC.set(qc);
+				}
+			} else {
 				this.highestCommittedQC.set(qc);
 			}
```

Comment:
Could be simplified:
```suggestion
			if (!highest.isPresent() || highest.get().getView().compareTo(vertexMetadata.getView()) < 0) {
				this.highestCommittedQC.set(qc);
			}
```

Reply:
cool